xarray/backends/api.py

```diff
@@ -870,13 +872,27 @@ def open_mfdataset(
     """
     if isinstance(paths, str):
         if is_remote_uri(paths):
-            raise ValueError(
-                "cannot do wild-card matching for paths that are remote URLs: "
-                "{!r}. Instead, supply paths as an explicit list of strings.".format(
-                    paths
-                )
+            from fsspec import open_files
+            from fsspec.core import get_fs_token_paths
+
+            so = kwargs.get("backend_kwargs", {}).get("storage_options", {})
+            # zarr requires mappers, other backends may work with file-like objects
+            # get_fs_token_paths also allows arguments embedded in URLs
+            fs, _, _ = get_fs_token_paths(
+                paths,
+                mode="rb",
+                storage_options=so,
+                expand=False,
             )
-        paths = sorted(glob(_normalize_path(paths)))
+            if engine == "zarr":
+                paths = fs.glob(fs._strip_protocol(paths))
+                paths = [fs.get_mapper(path) for path in paths]
+            else:
+                if not paths.startswith("http"):
+                    # pass through HTTP unchanged for backward compatibility
```

Comment:
```suggestion
                if not paths.startswith("http"):
                    # pass through everything not starting with "http" unchanged for backward compatibility
```
Isn't the comment here contradicting the code?